_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,325 @@
+
+Enzyme was released in **2015 by AirBnB**. And let’s face it, Enzyme has gained a lot of popularity. Last year, 
+the library was moved from AirBnB space to external github space as a separate independent organization. This step was 
+intended to encourage the community to further develop the library.
+
+![commit diffrences](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/02.png)
+
+![commit diffrences](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/03.png)
+###### Source [Enzyme github](https://github.com/enzymejs/enzyme/commit/b06750bb5f6248ceb9c3fae903a71d0747a420d6)
+
+When it comes to raw data, the latest version of **Enzyme 3.11.0** has a bundle size of **463.2kb (127.5 kB)** and was 
+released on **20 December 2019**. Currently has 258 [open issues](https://github.com/enzymejs/enzyme/issues) and 31 
+open PR’s. 	
+
+![bundlephobia Enzyme sreen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/04.png)
+###### Source [Bundle Phobia](https://bundlephobia.com/package/enzyme@3.11.0)
+
+
+Interesting fact: 
+- There are also Enzyme Adapter Packages for React. So you need to install adapter package and its dependencies related 
+to the version of React that you are using. For example, for version 16, it also needs *enzyme—adapter—react—16* to be 
+installed. Unfortunately, it does not support all features completely. So for every new version of React, we have to 
+wait for the release of the new adapter.
+
+## React Testing Library (often called RTL)
+
+RTL says about itself: *The React Testing Library is a very light—weight solution for testing React components. It 
+provides light utility functions on top of react—dom and react—dom/test—utils, in a way that encourages better testing 
+practices.*.
+
+Released in **April 2018 by Kent C. Dodds**. The latest stable version **12.1.2** of RTL has a bundle size of 
+**215.8kb (47.3 kB)** and was released in **October 2021**. Currently, a pre—release of v13.0.0—alpha is being 
+prepared. Repository has 26 [open issues](https://github.com/enzymejs/enzyme/issues) and 2 open PR’s.  
+
+![bundlephobia RTL screen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/05.png)
+
+###### Source [Bundle Phobia](https://bundlephobia.com/package/@testing-library/react@12.1.2)
+
+
+Interesting facts: 
+- When you create a project by Create React App, you will get support for React Testing Library right away. 
+- Jest documentation links to the RTL website in the [*Testing Web Framework*](https://jestjs.io/docs/testing-frameworks#react)
+ section as well as React documentation in the [*Recommended Tools*](https://reactjs.org/docs/testing.html#tools).
+- Testing—library allows us to test with the same API not only React but also others frameworks like Vue 
+(testing—library/vue), Angluar (tesiting—library/angular), Svelte (testing—library/svelte) etc. This could be some 
```

Comment:
there's also dom-testing-library which is used by all four of them and allows writing tests without any framework whatsoever!